packages/core/src/shared/utilities/cacheUtils.ts

```diff
@@ -131,14 +130,12 @@ export function createDiskCache<T, K>(
                 await fs.mkdir(dirname(target))
                 if (isWeb()) {
                     // There is no web-compatible rename() method. So do a regular write.
-                    await fs.writeFile(target, JSON.stringify(data), { mode: 0o600 })
+                    await fs.writeFile(target, JSON.stringify(data))
```

Comment:
Since we are using the VS Code File system implementation which does not support setting mode/permissions, we do not need this anymore

Reply:
We do need it. See https://github.com/aws/aws-toolkit-vscode/pull/5361/files#r1688216497

This PR removed functionality. We need to be able to specify permissions where possible. That may require special-casing fs.writeFile internally, when nodejs is available.

Reply:
Because this is the `isWeb` case. But not for the "nodejs" case.